analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -175,37 +172,42 @@ class Analyzer(private val config: AnalyzerConfiguration, private val labels: Ma
 
     private fun determinePackageManagerDependencies(
         managedFiles: Map<PackageManager, List<File>>
-    ): Map<PackageManager, Set<PackageManager>> {
-        val allPackageManagers =
-            managedFiles.keys.associateBy { it.managerName }.toSortedMap(String.CASE_INSENSITIVE_ORDER)
+    ): Map<PackageManager, Set<String>> {
+        val packageManagersWithFiles = managedFiles.keys.associateByTo(sortedMapOf(String.CASE_INSENSITIVE_ORDER)) {
+            it.managerName
+        }
 
-        val result = mutableMapOf<PackageManager, MutableSet<PackageManager>>()
+        val result = mutableMapOf<PackageManager, MutableSet<String>>()
 
         managedFiles.keys.forEach { packageManager ->
             val dependencies = packageManager.findPackageManagerDependencies(managedFiles)
-            dependencies.mustRunAfter.forEach { name ->
-                val managerForName = allPackageManagers[name]
+
+            val mustRunAfter = config.getPackageManagerConfiguration(packageManager.managerName)?.mustRunAfter?.toSet()
+                ?: dependencies.mustRunAfter
```

Comment:
There might still be a problem with this solution as now a dynamic `mustRunBefore` (see below) may override an inverse `mustRunAfter` from the static / global configuration, which might not be what we want. I need to clarify this with @mnonnenmacher.

Reply:
I've added a fixup commit that I'll eventually squash later that shows how this could be addressed.

Reply:
Probably yes, but that filter would need to be done on the `config`-retrieved dependencies only, as `determinePackageManagerDependencies()` and thus the `packageManagerDependencies` map already have that filtering applied. And as `determinePackageManagerDependencies()` already does that filtering, I wanted to rather make reuse of it instead of implementing it again. Also, reusing `determinePackageManagerDependencies()` has the benefit to also get the debug log output when a package manager dependency is ignored.